src/logic/basic.lean

```diff
@@ -66,6 +66,9 @@ by cc
 lemma subsingleton_of_forall_eq {α : Sort*} (x : α) (h : ∀ y, y = x) : subsingleton α :=
 ⟨λ a b, (h a).symm ▸ (h b).symm ▸ rfl⟩
 
+lemma subsingleton_iff_forall_eq {α : Sort*} (x : α) : subsingleton α ↔ ∀ y, y = x :=
+⟨λ h y, @subsingleton.elim _ h y x, λ h, subsingleton_of_forall_eq x h⟩
```

Comment:
golf:
```suggestion
⟨λ h y, @subsingleton.elim _ h y x, subsingleton_of_forall_eq x⟩
```
but not worth delaying bors if its already running

Reply:
Ops, I did not know that cancelled bors

Reply:
@bryangingechen Sorry, can you start bors again

Reply:
No problem! Taking a PR off the queue doesn't slow down the merging unless bors has already pushed a commit to the `staging` branch.
bors r+
(just in case...)
bors d+